### PR TITLE
Block lens

### PR DESCRIPTION
**Pull request by vext01**
This change exposes the lengths of the PT blocks via hwtracer. We use this to attempt to rake back traced MIR blocks when the MIR block doesn't start at the same address as the PT block.

Also kill some tests (see the commit message).

Cheers.

### CODE REVIEW

File: src/backends/perf_pt/decode.c [hunk 1]
```diff
@@ -169,20 +170,20 @@ perf_pt_init_block_decoder(void *buf, uint64_t len, int vdso_fd, char *vdso_file
 }
 
 /*
- * Updates `*addr' with the start address of the next block in the instruction
- * stream.
+ * Updates `*addr' and `*len' with the start address and the length of of the
```
[the remark this answers is not in the document]
Comment (vext01) [reply src/backends/perf_pt/decode.c]: I think at the time of writing that was intentional, but I agree that since this is markdown, we want backticks.

**Comment by vext01** [comment]
Sorry, i missed the "of of" thing.

**Comment by vext01** [comment]
Fixed. Ready for re-review.

**Comment by ltratt** [comment]
Please squash.

**Comment by ltratt** [comment]
Could I ask you to motivate in the commit message why we need to expose the block lengths?

**Comment by vext01** [comment]
Done.

**Pull request merged**